cloud-controller-manager/do/health.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2017 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/digitalocean/godo"
+)
+
+var godoHealthTimeout = 15 * time.Second
+
+type godoHealthChecker struct {
+	client *godo.Client
+}
+
+func (c *godoHealthChecker) Name() string {
+	return "godo"
+}
+
+func (c *godoHealthChecker) Check(r *http.Request) error {
+	ctx, cancel := context.WithTimeout(r.Context(), godoHealthTimeout)
+	defer cancel()
+	_, _, err := c.client.Account.Get(ctx)
```

Comment:
If we use the default kubernetes health-check duration (10s) we'll see 360 req/hour, which is a non-trivial portion of the 5000 req/hour limit.

Reply:
I think this health check won't actually get triggered by that health checking mechanism unless we (or the user) intentionally configures it right?

My thinking here is our agent would be the only thing that hits the endpoint, and that would be triggered once every few minutes by our agent monitor service in addition to once every reconcile.

The diy cluster case I'm not too sure about though; do you think clear documentation around this is sufficient?

Reply:
Yeah, that's fair.  I also forgot that request limit is scoped to the token, not the whole account.